pkg/manifest/ostree_deployment.go

```diff
@@ -164,7 +171,7 @@ func (p *OSTreeDeployment) serialize() osbuild.Pipeline {
 		&osbuild.OSTreeDeployStageOptions{
 			OsName: p.osName,
 			Ref:    commit.Ref,
-			Remote: p.Remote.Name,
+			Remote: remoteName,
```

Comment:
This is the cause of the problem with the upgrades (see https://github.com/achilleas-k/fedora-iot-remote).  When the remote configuration is coming through the package, we don't specify a remote name so the deployment doesn't have a configured remote.  And even adding one, with `ostree remote add ...`, it doesn't change the configuration of the deployment.

Compare
```
$ cat /ostree/deploy/fedora-iot/deploy/22543d2fc8d14087616f07ccf48d75eab535c26e81ecb31f4adf4d70f313f923.0.origin
[origin]
refspec=fedora/39/x86_64/iot
```
with
```
$ cat /ostree/deploy/fedora-iot/deploy/22543d2fc8d14087616f07ccf48d75eab535c26e81ecb31f4adf4d70f313f923.0.origin
[origin]
refspec=fedora-iot:fedora/39/x86_64/iot
```

Reply:
CC @henrywang